src/layers/tile/util.rs

```diff
@@ -10,18 +10,24 @@ pub(crate) fn parse_data_line(
     parser: &mut impl Iterator<Item = XmlEventResult>,
     tilesets: &[MapTilesetGid],
 ) -> Result<Vec<Option<LayerTileData>>, TiledError> {
-    match (encoding, compression) {
+    match (encoding.as_deref(), compression.as_deref()) {
         (None, None) => {
-            return Err(TiledError::Other(
-                "XML format is currently not supported".to_string(),
-            ))
+            return Err(TiledError::InvalidEncodingFormat {
+                encoding,
+                compression,
+            })
         }
-        (Some(e), None) => match e.as_ref() {
-            "base64" => return parse_base64(parser).map(|v| convert_to_tiles(&v, tilesets)),
-            "csv" => return decode_csv(parser, tilesets),
-            e => return Err(TiledError::Other(format!("Unknown encoding format {}", e))),
-        },
-        (Some(e), Some(c)) => match (e.as_ref(), c.as_ref()) {
+        (Some("base64"), None) => {
+            return parse_base64(parser).map(|v| convert_to_tiles(&v, tilesets))
+        }
+        (Some("csv"), None) => return decode_csv(parser, tilesets),
+        (Some(_), None) => {
+            return Err(TiledError::InvalidEncodingFormat {
+                encoding,
+                compression,
+            })
+        }
+        (Some(e), Some(c)) => match (e, c) {
```

Comment:
I would suggest to use `(Some("base64"), c)` here, cause then we only need to match on `c` afterwards and can merge the `(Some("base64"), None)` case in here, and have only a single place where we call to `parse_base64`.

Reply:
I know, but since this was a PR for errors I decided not to fix it until later